Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -1696,275 +1680,381 @@ namespace OpenInfraPlatform
+				//	END_ENTITY;
+				// **************************************************************************************************************************
 
-					double radius = (typename IfcEntityTypesT::IfcLengthMeasure)(sphere->Radius);
+				std::shared_ptr<carve::input::PolyhedronData> polyhedronData(new carve::input::PolyhedronData());
+				carve::math::Matrix primitivePlacementMatrix = sphere->Position ?
+					pos * placementConverter->convertIfcAxis2Placement3D(sphere->Position) :
+					pos;
 
-					// seen from the top, each ring is:
-					//        \   |   /
-					//         2- 1 -nvc
-					//        / \ | / \
-					//    ---3--- 0 ---7---
-					//       \  / | \ /
-					//         4- 5 -6
-					//        /   |   \
+				double radius = sphere->Radius * UnitConvert()->getLengthInMeterFactor();
 
-					std::shared_ptr<carve::input::PolyhedronData> polyhedron_data(new carve::input::PolyhedronData());
-					polyhedron_data->addVertex(pos*carve::geom::VECTOR(0.0, 0.0, radius)); // top
+				// seen from the top, each ring is:
+				//        \   |   /
+				//         2- 1 -numberOfVertices
+				//        / \ | / \
+			    //    ---3--- 0 ---7---
+				//       \  / | \ /
+				//         4- 5 -6
+				//        /   |   \
 
-					const int nvc = GeomSettings()->getNumberOfSegmentsForTessellation(radius);
-					const double d_horizontal_angle = GeomSettings()->getAngleLength(radius);
-					const int num_vertical_edges = ceil(0.5 * nvc);
-					double d_vertical_angle = M_PI / double(num_vertical_edges - 1);	// TODO: adapt to model size and complexity
-					double vertical_angle = d_vertical_angle;
 
-					for (int vertical = 1; vertical < num_vertical_edges - 1; ++vertical)
-					{
-						// for each vertical angle, add one horizontal circle
-						double vertical_level = cos(vertical_angle)*radius;
-						double radius_at_level = sin(vertical_angle)*radius;
-						double horizontal_angle = 0;
-						for (int i = 0; i < nvc; ++i)
-						{
-							polyhedron_data->addVertex(primitive_placement_matrix*carve::geom::VECTOR(sin(horizontal_angle)*radius_at_level, cos(horizontal_angle)*radius_at_level, vertical_level));
-							horizontal_angle += d_horizontal_angle;
-						}
-						vertical_angle += d_vertical_angle;
-					}
-					polyhedron_data->addVertex(primitive_placement_matrix*carve::geom::VECTOR(0.0, 0.0, -radius)); // bottom
+				polyhedronData->addVertex(pos*carve::geom::VECTOR(0.0, 0.0, radius)); // top
 
-					// uppper triangle fan
-					for (int i = 0; i < nvc - 1; ++i)
-					{
-						polyhedron_data->addFace(0, i + 2, i + 1);
-					}
-					polyhedron_data->addFace(0, 1, nvc);
+				const int numberOfVertices = GeomSettings()->getNumberOfSegmentsForTessellation(radius);
+				const double horizontalAngleDelta = GeomSettings()->getAngleLength(radius);
+				const int num_vertical_edges = ceil(0.5 * numberOfVertices);
+				double verticalAngleDelta = M_PI / double(num_vertical_edges - 1);	// TODO: adapt to model size and complexity

```

Comment:
this todo is done, by using ->getNumberOfSegmentsForTessellation - the note can be removed

Reply:
Resolved: 207de31